store/storeIface.go

```diff
@@ -78,4 +78,12 @@ type Store interface {
 	SaveBlockCid(height uint64, cid cid.Cid, batch KVBatch) (KVBatch, error)
 
 	LoadBlockCid(height uint64) (cid.Cid, error)
+
+	SaveBlockSource(height uint64, source string, batch KVBatch) (KVBatch, error)
+
+	LoadBlockSource(height uint64) (string, error)
+
+	SaveValidationHeight(height uint64, batch KVBatch) (KVBatch, error)
+
+	LoadValidationHeight() (uint64, error)
```

Comment:
what is the reason for having this in store AND also in memory on the uint64?
it's a bit confusing

Reply:
block validation needs to fetch source from all blocks stored. right now block source was only added in the (in memory) block cache, but validation dont get blocks from that cache.